### ISSUE TITLE
Stats: outbound links to Substack not being tracked

### ISSUE THREAD

**Issue by liviopv** [Automattic/wp-calypso#72226]
### Quick summary

Clicks in outbound links to https://substack.com URLs are not being tracked

### Steps to reproduce

1. Add an outbound link to a https://substack.com URL to a page or post
2. Open the page in an incognito window
3. Click on the links to Substack

### What you expected to happen

The Substack URL should be listed under the "Clicks" section - https://wordpress.com/stats/day/clicks/

### What actually happened

The Substack URL doesn't show up on https://wordpress.com/stats/day/clicks/ even thought visitors and view counts were updated correctly.

### Context

User report:

- [ ] 5860682-zd-woothemes

### Browser, operating system and other notes

Tests were conducted on Chrome. Cookies collection was accepted before clicking the links

### Reproducibility

Consistent

### Severity

Some (< 50%)

### Available workarounds?

No but the platform is still usable

**Comment by dpasque** [comment]
I was troubleshooting this on both a Jurassic Ninja site and WPCOM Simple Site, and in both cases, I was able to get stats to trigger for both https://substack.com, and a specific substack (with a subdomain).

I loaded the posts in an incognito window to emulate being someone from the public (and not the author), and on both sites, I got stats to trigger for both URLs.

I wonder if there might be something I'm missing in the steps to reproduce?

@liviopv -- any extra guidance you might have? Are you still able to reproduce the behavior?
